### ISSUE TITLE
Omitting VSPHERE_SSH_AUTHORIZED_KEY creates nil/null error for deploying workload clusters

### ISSUE THREAD

**Issue by kacole2** [kubernetes-sigs/cluster-api-provider-vsphere#800]
/kind bug

**What steps did you take and what happened:**
Following the getting [started guide](https://github.com/kubernetes-sigs/cluster-api-provider-vsphere/blob/master/docs/getting_started.md) for the latest wit v1alpha3. I'm leaving `VSPHERE_SSH_AUTHORIZED_KEY:` set to `""` and not passing a SSH public key.

I'm creating the cluster spec with:
```
clusterctl config cluster vsphere-quickstart \
    --infrastructure vsphere:v0.6.0-rc.1 \
    --kubernetes-version v1.17.3 \
    --control-plane-machine-count 1 \
    --worker-machine-count 3 > cluster.yaml
```

Then deploying with:
```
kubectl apply -f cluster.yaml
```

The error received is:
```
vmware@kcjump:~$ kubectl apply -f cluster.yaml
cluster.cluster.x-k8s.io/vsphere-quickstart created
error: error validating "cluster.yaml": error validating data: ValidationError(HAProxyLoadBalancer.spec.user.authorizedKeys): unknown object type "nil" in HAProxyLoadBalancer.spec.user.authorizedKeys[0]; if you choose to ignore these errors, turn validation off with --validate=false
```

I tried adding `--validate=false` and received:
```
vmware@kcjump:~$ kubectl apply -f cluster.yaml --validate=false
cluster.cluster.x-k8s.io/vsphere-quickstart unchanged
vspherecluster.infrastructure.cluster.x-k8s.io/vsphere-quickstart created
vspheremachinetemplate.infrastructure.cluster.x-k8s.io/vsphere-quickstart created
machinedeployment.cluster.x-k8s.io/vsphere-quickstart-md-0 created
Error from server (Invalid): error when creating "cluster.yaml": HAProxyLoadBalancer.infrastructure.cluster.x-k8s.io "vsphere-quickstart" is invalid: spec.user.authorizedKeys: Invalid value: "null": spec.user.authorizedKeys in body must be of type string: "null"
Error from server (Invalid): error when creating "cluster.yaml": KubeadmControlPlane.controlplane.cluster.x-k8s.io "vsphere-quickstart" is invalid: spec.kubeadmConfigSpec.users.sshAuthorizedKeys: Invalid value: "null": spec.kubeadmConfigSpec.users.sshAuthorizedKeys in body must be of type string: "null"
Error from server (Invalid): error when creating "cluster.yaml": KubeadmConfigTemplate.bootstrap.cluster.x-k8s.io "vsphere-quickstart-md-0" is invalid: spec.template.spec.users.sshAuthorizedKeys: Invalid value: "null": spec.template.spec.users.sshAuthorizedKeys in body must be of type string: "null"
```

After I added a SSH public key to the configuration file the creation process started as expected.

**What did you expect to happen:**

Deploy clusters without specifying 

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]

**Environment:**

- Cluster-api-provider-vsphere version: v1alpha3
- Kubernetes version: (use `kubectl version`): 1.17.3
- OS (e.g. from `/etc/os-release`):  Ubuntu 18.04

**Comment by randomvariable** [comment]
Limitation in clusterctl at present. See https://github.com/kubernetes-sigs/cluster-api/issues/2497

CC @fabriziopandini

**Comment by fabriziopandini** [comment]
@kacole2 @randomvariable the fix on clusterctl for allowing empty env variables is merged

**Comment by yastij** [comment]
closing this since it's fixed in v0.3.0
